lib/src/common/services/moodle_managers.dart

```diff
@@ -87,29 +87,42 @@ class MoodleCourseManager with ChangeNotifier {
           sortedCourses.where((c) => c.customFavorite ?? false).toList();
     } else if (filterBy == MoodleCourseFilteringType.byLatestSemester) {
       int? latestYear, latestSemester;
-      Map<num, List<int>> coursesYearAndSemester = {};
+      Map<num, List<int?>> coursesYearAndSemester = {};
 
       for (final course in sortedCourses) {
-        var groups =
-            RegExp(r'^(.*)_(\w{0,3})_(\d{4})$').firstMatch(course.idnumber);
+        int? year, sem;
+        String? yearString, semString;
+
+        // One semester course
+        final oneSemesterGroups =
+            RegExp(r'^(.*)_(\w{1,3})_(\d{4})$').firstMatch(course.idnumber);
+        if (oneSemesterGroups != null) {
```

Comment:
Another corner case: _ABCD_ALL_2024 (which is a full-year course)
Better check if semString is valid as well

Reply:
I think it is already handled by the "int.tryParse". Or should I check if the sem is between 1 to 3?

Reply:
Ah I see. Makes sense